packages/excalidraw/tests/history.test.tsx

```diff
@@ -1845,6 +1858,20 @@ describe("history", () => {
       expect(queryByTestId(container, "button-undo")).toBeDisabled();
       expect(queryByTestId(container, "button-redo")).not.toBeDisabled();
     });
+
+    it("should record history when pasting an image", async () => {
+      await render(
+        <Excalidraw autoFocus={true} handleKeyboardGlobally={true} />,
+      );
+      pasteImage();
+      await waitFor(() => {
+        expect(API.getUndoStack().length).toBe(1);
+        expect(API.getRedoStack().length).toBe(0);
+        Keyboard.undo();
```

Comment:
We should likely have multiple tests just for the simple paste like:
- pasting multiple images in a row
- performing other sync actions while the image is loading

and we should also extend each such test with additional checks, like:
- check the elements and their expected properties after applying the changes, not just that there is something unknown in the stack
- iterate the stack multiple times (undo, redo, undo), and on each check everything again